### PR TITLE
Create Dockerfile and app-interface scripts

### PR DESCRIPTION
**Pull request by chambridge**
* Create Dockerfile to build image
* Add app-interface build and pr check scripts

https://issues.redhat.com/browse/COST-1399

### CODE REVIEW

File: build_deploy.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+IMAGE_TAG=$(git rev-parse --short=7 HEAD)
+
+if [[ -z "$QUAY_USER" || -z "$QUAY_TOKEN" ]]; then
+    echo "QUAY_USER and QUAY_TOKEN must be set"
+    exit 1
+fi
+
+DOCKER_CONF="$PWD/.docker"
+mkdir -p "$DOCKER_CONF"
+docker --config="$DOCKER_CONF" login -u="$QUAY_USER" -p="$QUAY_TOKEN" quay.io
```
Comment (adberglund) [inline build_deploy.sh:19]: does this need to be podman now?

Comment (chambridge) [reply build_deploy.sh:19]: I saw https://github.com/project-koku/koku/pull/2956

Doesn't seem to be effecting other teams builds:
https://ci.ext.devshift.net/job/RedHatInsights-edge-api-gh-build-main/43/console
https://github.com/RedHatInsights/edge-api/blob/main/build_deploy.sh

Seems like it was related to this in the JJB definition:
```
    label: rhel8
    node: jenkins-slave-09-rhel8
```

Is this required for s2i tooling or something? @maskarb

Comment (maskarb) [reply build_deploy.sh:19]: That node gives us access to py38 which we would need for running unittests (if we were to switch to using ci-ext to run unittests). If there is no need for py38, then we don't need to use the `node: jenkins-slave-09-rhel8`.

Comment (chambridge) [reply build_deploy.sh:19]: ok. so we should be fine for the repo then.

**Review by adberglund — APPROVED** [review]

**Pull request merged**